jsparagus/emit/rust.py

```diff
@@ -200,18 +218,33 @@ def write_condition(self, state, first_act):
             if len(state.epsilon) == 1:
                 # This is an attempt to avoid huge unending compilations.
                 _, dest = next(iter(state.epsilon), (None, None))
-                self.write("// parser.top_state() in [{}]", " | ".join(map(str, first_act.states)))
+                pattern = rust_range(extract_ranges(first_act.states))
+                self.write("// parser.top_state() in ({})", pattern)
                 self.write_epsilon_transition(dest)
             else:
                 self.write("match parser.top_state() {")
                 with indent(self):
+                    max_weight = max(len(act.states) for act, dest in state.edges())
+                    max_states = []
+                    max_dest = None
                     for act, dest in state.edges():
                         assert first_act.check_same_variable(act)
-                        self.write("{} => {{", " | ".join(map(str, act.states)))
+                        if max_dest is None and max_weight == len(act.states):
```

Comment:
what does this condition mean?

Reply:
(I will add the following as a comment)

This condition is used to make the largest range the default condition. The `top_state` is guaranteed to be in the ranges which are matched. Thus instead of having a default case which would panic, as it is unreachable, we use the default case as a way to represent the most likely target to continue with.

This should be a minor speed improvement, as we would not produce useless code, and as the default branch becomes the likely branch as well.